DSharpPlus/Clients/DiscordClient.cs

```diff
@@ -838,38 +839,49 @@ public async Task<DiscordApplicationCommand> EditGuildApplicationCommandAsync(ul
     
     
     /// <summary>
-    /// Returns a list of guilds before a certain guild.
+    /// Returns a list of guilds before a certain guild. One api request per 200 guilds
     /// <param name="limit">The amount of guilds to fetch.</param>
     /// <param name="before">The ID of the guild before which we fetch the guilds</param>
     /// <param name="withCount">Whether to include approximate member and presence counts in the returned guilds.</param>
+    /// <param name="cancellationToken">Cancels the enumeration before doing the next api request</param>
     /// </summary>
     /// <exception cref="BadRequestException">Thrown when an invalid parameter was provided.</exception>
     /// <exception cref="ServerErrorException">Thrown when Discord is unable to process the request.</exception>
-    public IAsyncEnumerable<DiscordGuild> GetGuildsBeforeAsync(ulong before, int limit = 200, bool? withCount = null)
-        => this.GetGuildsInternalAsync(limit, before, null, null);
+    public IAsyncEnumerable<DiscordGuild> GetGuildsBeforeAsync(ulong before, int limit = 200, bool? withCount = null, CancellationToken cancellationToken = default)
```

Comment:
Consider also annotating these with `[EnumeratorCancellation]`

Reply:
Analyzer said it would be without effect